sktime/datatypes/_adapter/polars.py

```diff
@@ -119,6 +143,47 @@ def check_polars_frame(obj, return_metadata=False, var_name="obj", lazy=False):
         return ret(False, msg, None, return_metadata)
 
     # we now know obj is a polars DataFrame or LazyFrame
+    index_cols = get_mi_cols(obj)
+
+    if scitype == "Series":
+        cols_msg = (
+            f"{var_name} must have exactly one index column, "
+            f"found {len(index_cols)}, namely: {index_cols}"
+        )
+        right_no_index_cols = len(index_cols) <= 1
```

Comment:
use a dictionary instead of long `if` / `elif` chain with substantial code duplication.

Reply:
yess, good idea!

Reply:
for dictionary solution I'm thinking of something that will look like this 
```python
scitypes = {
        "Series": len(index_cols) == 1,
        "Panel": len(index_cols) == 2,
        "Hierarchical": len(index_cols) >= 3,
    }
if not scitypes[scitype]:
      cols_msg = (
          f"{var_name} must have correct number of index columns for scitype,"
          f"Series: 1, Panel: 2, Hierarchical: >= 3,"
          f"found {len(index_cols)}, namely: {index_cols}"
      )
      return ret(False, cols_msg, None, return_metadata)
```